internal/controllers/clusterextension_controller.go

```diff
@@ -217,7 +216,7 @@ func (r *ClusterExtensionReconciler) reconcile(ctx context.Context, ext *ocv1alp
 	resolvedBundle, resolvedBundleVersion, resolvedDeprecation, err := r.Resolver.Resolve(ctx, ext, installedBundle)
 	if err != nil {
 		// Note: We don't distinguish between resolution-specific errors and generic errors
-		setInstallStatus(ext, nil)
+		setInstalledStatusFromBundle(ext, installedBundle, err)
```

Comment:
Since we successfully got the installed bundle (or know there isn't an installed bundle), it seems like there shouldn't be any error messages in the `Installed` condition. It would be one of:

```yaml
- type: Installed
  status: True
  reason: Success
  # This is interesting because I don't think we store the reference in the 
  # installed bundle metadata, maybe we should so that the message is
  # able to be consistently recreated.  Just need to be careful not to assume
  # the reference is an image because it could (in the future) come from
  # somewhere else.
  message: "Installed bundle <bundleImageRef> successfully"
```

Or:
```yaml
- type: Installed
  status: False
  reason: Failed
  message: "no installed bundle found"
```

So should be pass a `nil` error here or `errors.New("no installed bundle found")`?

Reply:
Similar comment for other places that we call this function after a successful call to `GetInstalledBundle`